### PR TITLE
Add inspection for Mixin public static members

### PR DESCRIPTION
**Pull request by Aaron1011**

### CODE REVIEW

File: src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+package com.demonwav.mcdev.platform.mixin.inspections;
```
Comment (kashike) [inline src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java]: Missing licence header.

Comment (Aaron1011) [inline src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java]: Fixed

File: src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ *
+ * Copyright (c) 2016 Kyle Wood (DemonWav)
+ *
+ * MIT License
+ */
+package com.demonwav.mcdev.platform.mixin.inspections;
+
+import com.demonwav.mcdev.platform.mixin.util.MixinConstants;
+import com.demonwav.mcdev.platform.mixin.util.MixinUtils;
+import com.demonwav.mcdev.util.McPsiUtil;
```
Comment (kashike) [inline src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java]: Wrong import layout.

File: src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        throw new IllegalArgumentException("Cannot build error message for PsiMember " + member);
+    }
+
+    private String getFieldMessage(PsiMember field) {
+        return String.format("Static field '%s' is part of a Mixin, and therefore must not be public", field.getName());
+    }
+
+    private String getMethodMessage(PsiMember method) {
+        return String.format("Non-Overwrite static method %s is part of a Mixin, and therefore must not be public", method.getName());
```
Comment (DemonWav) [inline src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java]: `%s` should be surrounded with `'`s, like the field error message.

File: src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+            @Override
+            public void visitField(PsiField field) {
+                if (isProblematic(field)) {
+                    this.registerError(field, field);
+                }
+            }
+
+            private boolean isProblematic(PsiMember member) {
+                PsiClass mixin = MixinUtils.getContainingMixinClass(member);
```
Comment (DemonWav) [inline src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java]: Should be final.

File: src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                if (mixin == null) {
+                    return false;
+                }
+
+                PsiModifierList modifierList = mixin.getModifierList();
```
Comment (DemonWav) [inline src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java]: Should be final.

File: src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+                if (modifierList == null) {
+                    return false;
+                }
+
+                boolean isOverwrite = member instanceof PsiMethod && McPsiUtil.getAnnotation(member, MixinConstants.Annotations.OVERWRITE) != null;
```
Comment (DemonWav) [inline src/main/java/com/demonwav/mcdev/platform/mixin/inspections/StaticMixinMemberInspection.java]: Should be final.

**Pull request merged**